### PR TITLE
Add sigmoid cross entropy function

### PR DESCRIPTION
**Pull request by rezoo**
This PR aims to implement a sigmoid cross entropy loss function which is already implemented in caffe.
The sigmoid cross entropy loss function would be useful when we construct a network whose output layer consists of a set of binary classifiers.

### CODE REVIEW

**Comment by beam2d** [comment]
Thank you for the PR! We will review it for the next minor release, so wait a little.

**Comment by rezoo** [comment]
I understand. Anyway, thanks for releasing such a cool library :)

**Comment by beam2d** [comment]
Sorry for being late. I will review it. Thank you very much for following the master updates (especially the style guidelines)!

File: chainer/functions/sigmoid_cross_entropy.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            x * (t - (x >= 0)) - numpy.log1p(numpy.exp(-numpy.abs(x))))
+        return numpy.array((loss / t.shape[0],), dtype=numpy.float32),
+
+    def forward_gpu(self, inputs):
+        x, t = inputs
+        self.y, = sigmoid.Sigmoid(self.use_cudnn).forward_gpu((x,))
+        loss = -cuda.reduce(
+            'int* t, float* x',
+            'x[i] * (t[i] - (x[i] >= 0))'
+            ' - log1pf(expf(x[i] - 2 * x[i] * (x[i] >= 0)))',
```
Comment (beam2d) [inline chainer/functions/sigmoid_cross_entropy.py]: How about using a triple-quote string for readability?

Comment (beam2d) [inline chainer/functions/sigmoid_cross_entropy.py]: Use [fabsf](http://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html#group__CUDA__MATH__SINGLE_1g2df1125a453c1ea3b294dd288ed455b0).

File: chainer/functions/sigmoid_cross_entropy.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            'sigmoid_crossent_bwd')(gx, self.y, t, coeff)
+        return gx, None
+
+
+def sigmoid_cross_entropy(x, t, use_cudnn=True):
+    """Computes cross entropy loss for sigmoid activations.
+
+    Args:
+        x (Variable): Variable holding a matrix whose (i, j)-th element
+            indicates unnormalized log probability of the j-th unit at
```
Comment (beam2d) [inline chainer/functions/sigmoid_cross_entropy.py]: indicates **the** unnormalized ...

File: chainer/functions/sigmoid_cross_entropy.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            'gx[i] = *coeff * (y[i] - t[i])',
+            'sigmoid_crossent_bwd')(gx, self.y, t, coeff)
+        return gx, None
+
+
+def sigmoid_cross_entropy(x, t, use_cudnn=True):
+    """Computes cross entropy loss for sigmoid activations.
+
+    Args:
+        x (Variable): Variable holding a matrix whose (i, j)-th element
```
Comment (beam2d) [inline chainer/functions/sigmoid_cross_entropy.py]: Variable -> A variable object

File: chainer/functions/sigmoid_cross_entropy.py (hunk 2 above)
Comment (beam2d) [inline chainer/functions/sigmoid_cross_entropy.py]: indicates **the** unnormalized ...

File: chainer/functions/sigmoid_cross_entropy.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            indicates unnormalized log probability of the j-th unit at
+            the i-th example.
+        t (Variable): Variable holding an int32 vector of groundtruth
```
Comment (beam2d) [inline chainer/functions/sigmoid_cross_entropy.py]: Variable -> A variable object

File: chainer/functions/sigmoid_cross_entropy.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            binary labels.
+
+    Returns:
+        Variable: A variable holding a scalar array of the cross entropy loss.
```
Comment (beam2d) [inline chainer/functions/sigmoid_cross_entropy.py]: A variable **object** holding ...

File: tests/functions_tests/test_sigmoid_cross_entropy.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        t_val = chainer.Variable(t_data)
+        loss = functions.sigmoid_cross_entropy(x_val, t_val, use_cudnn)
+        loss_value = float(cuda.to_cpu(loss.data))
+
+        # Compute expected value
+        loss_expect = 0
+        for i in six.moves.range(self.x.shape[0]):
+            for j in six.moves.range(self.x.shape[1]):
+                xd, td = self.x[i, j], self.t[i, j]
+                xdd = 1 if self.x[i, j] >= 0 else 0
```
Comment (beam2d) [inline tests/functions_tests/test_sigmoid_cross_entropy.py]: You can use a bool directly as a 0/1 integer: `xdd = self.x[i, j] >= 0`. If you want to convert it into a 0/1 int explicitly, use int() instead of the if expression.

**Comment by beam2d** [comment]
LGTM except above small comments!

**Comment by rezoo** [comment]
Thanks for your comments! I'll fix them.

**Comment by beam2d** [comment]
Thank you!
